src/custom_source_buffers/text/html/html_text_source_buffer.ts

```diff
@@ -291,47 +290,57 @@ export default class HTMLTextSourceBuffer
   /**
    * Remove the current cue from being displayed.
    */
-  private _hideCurrentCue() : void {
+  private _disableCurrentCues() : void {
     this._clearSizeUpdates$.next();
-    if (this._currentCue !== null) {
-      safelyRemoveChild(this._textTrackElement, this._currentCue.element);
-      this._currentCue = null;
+    if (this._currentCues.length > 0) {
+      for (let i = 0; i < this._currentCues.length; i++) {
+        safelyRemoveChild(this._textTrackElement, this._currentCues[i].element);
+      }
+      this._currentCues = [];
     }
   }
 
   /**
    * Display a new Cue. If one was already present, it will be replaced.
    * @param {HTMLElement} element
    */
-  private _displayCue(element : HTMLElement) : void {
-    if (this._currentCue !== null && this._currentCue.element === element) {
-      return; // we're already good
+  private _displayCues(elements : HTMLElement[]) : void {
+    const nothingChanged = this._currentCues.length === elements.length &&
+      this._currentCues.every((current, index) => current.element === elements[index]);
+
+    if (nothingChanged) {
+      return;
     }
 
+    // Remove and re-display everything
+    // TODO More intelligent handling
+
     this._clearSizeUpdates$.next();
-    if (this._currentCue !== null) {
-      safelyRemoveChild(this._textTrackElement, this._currentCue.element);
+    for (let i = 0; i < this._currentCues.length; i++) {
+      safelyRemoveChild(this._textTrackElement, this._currentCues[i].element);
     }
 
-    const resolution = getElementResolution(element);
-    this._currentCue = { element, resolution };
-    if (resolution !== null) {
+    this._currentCues = [];
+    for (let i = 0; i < elements.length; i++) {
+      const element = elements[i];
+      const resolution = getElementResolution(element);
+      this._currentCues.push({ element, resolution });
+      this._textTrackElement.appendChild(element);
+    }
+
+    if (this._currentCues.some(cue => cue.resolution !== null)) {
```

Comment:
Maybe you could keep here the indexes of concerned currentCues, to reuse them later instead of running through the whole cue array.

Reply:
Will see what I can do better here.

Reply:
Done